resilience4j-ratelimiter/src/main/java/io/github/resilience4j/ratelimiter/RateLimiter.java

```diff
@@ -327,7 +343,19 @@ static void waitForPermission(final RateLimiter rateLimiter) {
 	 *
 	 * @return {@code long} amount of nanoseconds you should wait for reserved permission. if negative, it means you failed to reserve.
 	 */
-	long reservePermission();
+	default long reservePermission() {
+		return reservePermission(1);
+	}
+
+	/**
+	 * Reserves a permission of a given weight from this rate limiter and returns nanoseconds you should wait for it.
+	 * If returned long is negative, it means that you failed to reserve permission,
+	 * possibly your  {@link RateLimiterConfig#getTimeoutDuration()} is less then time to wait for permission.
+	 *
+	 * @param weight weight of permission - use for systems where 1 call != 1 permission
+	 * @return {@code long} amount of nanoseconds you should wait for reserved permission. if negative, it means you failed to reserve.
+	 */
+	long reservePermission(int weight);
```

Comment:
Can we rename `weight` to `permits`. Analogous to `https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Semaphore.html#acquire(int)`?